xfer/publisher.go

```diff
@@ -83,12 +94,73 @@ func (p HTTPPublisher) Publish(rpt report.Report) error {
 	return nil
 }
 
+// Stop implements Publisher
+func (p HTTPPublisher) Stop() {}
```

Comment:
Should we track the current request and do `http.Transport.CancelRequest(r)` here?